examples/stable_diffusion_v2/cldm/cldm.py

```diff
@@ -24,60 +23,47 @@
 import mindspore as ms
 import mindspore.nn as nn
 import mindspore.ops as ops
+from mindspore import Tensor
+from mindspore import dtype as mstype
+from mindspore import numpy as msnp
 
 _logger = logging.getLogger(__name__)
 
 
-class ControlledUnetModel(UNetModel):
-    def construct(self, x, timesteps=None, context=None, control=None, only_mid_control=False, **kwargs):
-        hs = []
-
-        # self.set_train(False)
-
-        t_emb = timestep_embedding(timesteps, self.model_channels, repeat_only=False)
-        emb = self.time_embed(t_emb)
-        h = x
-        for celllist in self.input_blocks:
-            for cell in celllist:
-                h = cell(h, emb, context)
-            hs.append(h)
-        for module in self.middle_block:
-            h = module(h, emb, context)
-
-        # TODO: only upper part was in do not need update gradients, not sure if set_train(True) needed here
-        if control is not None:
-            h += control.pop()
-
-        for celllist in self.output_blocks:
-            if only_mid_control or control is None:
-                h = self.cat((h, hs.pop()))
-            else:
-                h = self.cat([h, hs.pop() + control.pop()])
-
-            for cell in celllist:
-                h = cell(h, emb, context)
-
-        return self.out(h)
-
-
 class ControlnetUnetModel(UNetModel):
-    def __init__(self, control_stage_config, guess_mode=False, strength=1.0, *args, **kwargs):
+    def __init__(self, control_stage_config, guess_mode=False, strength=1.0, sd_locked=True, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+        if sd_locked:
+            for param in self.get_parameters():
+                param.requires_grad = False
+
+        # add controlnet init
         self.controlnet = instantiate_from_config(control_stage_config)
         self.control_scales = (
             [strength * (0.825 ** float(12 - i)) for i in range(13)] if guess_mode else ([strength] * 13)
         )
 
+        for param in self.controlnet.get_parameters():
+            print("INFO: controlnet param trainable: ", param.requires_grad)
+            break
```

Comment:
Remove debugging code

Reply:
Hi. I would change it to `logging.info` to satisfy the potential needs.

Reply:
I meant the `break` statement.

Reply:
Oh. Thank you. Done.